### PR TITLE
Add a new Goal component.

### PR DESCRIPTION
**Pull request by greinard**
## Overview

This branch adds a new Goal component.

The component supports the following configuration:
- `label` - The label to display next to the progress indicator.
- `targetValue` - The value at which the goal has been "met" or "completed".
- `maxValue` - The maximum value to use for the progress indicator (the denominator).
- `valueProvider` - A `SingleValueProvider<number>` implementation that provides a single numeric progress value.  Current implementations include static, random, and custom field value providers.
- `icon` - An optional FontAwesome icon override.  By default the `star` (solid) icon is used.
- `notStartedColor` - Optionally override the color used when no progress has been made toward the goal.
- `inProgressColor` - Optionally override the color used when the value is greater than 0, but still less than the `targetValue`.
- `completedColor` - Optionally override the color used when the value is greater than or equal to the `targetValue` (i.e. once the goal has been met).

Here are a few samples from the storybook:

achievements_screenshot

I will reset the snapshot version numbers prior to merging.

## Security

REMINDER: All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [x] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [x] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] These changes do not introduce any security risks, or any such risks have been properly mitigated.

- No new security risks.  This branch just adds a new UI component for visualizing progress toward a goal.

### Testing
- [ ] MyDataHelps iOS app
- [ ] MyDataHelps Android app
- [ ] [MyDataHelps website](mydatahelps.org)

### Documentation
- [x] I have added relevant Storybook updates to this PR as well.

## Release Notes

- **New Features**
	- Introduced a new `Goal` component for visualizing goal progress with a pie chart.
	- Added a new CSS file to enhance the styling of goal-related components.
	- Implemented a preview state feature for goals with various states (not started, in progress, complete, etc.).
	- Enhanced Storybook configuration for the `Goal` component, showcasing different scenarios.

- **Chores**
	- Updated module exports to include new components and types for better accessibility.

### CODE REVIEW

File: src/components/container/Goal/Goal.css
Comment (greinard) [reply src/components/container/Goal/Goal.css:44]: Since the surrounding visualization's size uses a fixed pixel size, I want the icon to be use a fixed pixel size as well.

File: package.json
Comment (reneefromhold) [inline package.json:3]: revert before merging

File: src/components/container/Goal/Goal.css
Comment (reneefromhold) [inline src/components/container/Goal/Goal.css:4]: Did this come from a design? It does not look balanced in the top corner in the story.

File: src/components/container/Goal/Goal.css
Comment (reneefromhold) [inline src/components/container/Goal/Goal.css:7]: not used

File: src/components/container/Goal/Goal.css
Comment (reneefromhold) [inline src/components/container/Goal/Goal.css:13]: not used

File: src/components/container/Goal/Goal.css
Comment (reneefromhold) [inline src/components/container/Goal/Goal.css:19]: not used

File: src/components/container/Goal/Goal.tsx
Comment (reneefromhold) [inline src/components/container/Goal/Goal.tsx:100]: this class is not in the .css

**Review by reneefromhold — COMMENTED** [review]
This feels more like a presentation than container component. 
For example, this could not be dropped into vb without a wrapper. Unless I am missing something?

File: src/helpers/single-value-providers.ts
Comment (greinard) [reply src/helpers/single-value-providers.ts]: Done.

File: src/components/container/Goal/Goal.tsx
Comment (greinard) [reply src/components/container/Goal/Goal.tsx:25]: I'm going to hold off on this.  It causes more properties than I want to be present in the storybook controls, which requires additional work to exclude them.

File: src/components/container/Goal/Goal.css
Comment (greinard) [reply src/components/container/Goal/Goal.css]: I've switched to `margin` and used a value of `16px` for both the sides and top.  This gives it better balance with respect to the whitespace on the sides and top without adding too much space between sequential goal components.

File: src/components/container/Goal/Goal.css
Comment (greinard) [reply src/components/container/Goal/Goal.css]: Good catch.  Removed.

File: src/components/container/Goal/Goal.css
Comment (greinard) [reply src/components/container/Goal/Goal.css]: Removed.

File: src/components/container/Goal/Goal.css
Comment (greinard) [reply src/components/container/Goal/Goal.css]: Removed.

File: src/components/container/Goal/Goal.tsx
Comment (greinard) [reply src/components/container/Goal/Goal.tsx:100]: I like to give most all `div` elements a style class even if I haven't had to add any CSS rules yet, just in case we end up needing to add some or if a user of this component needs to override something with CSS rules downstream.  If not present, the CSS selector for the `div` becomes more complex and fragile (i.e. something like `.mdhui-goal-info > div:last-child` instead of just `.mdhui-goal-name`).

**Comment by greinard** [comment]
> This feels more like a presentation than container component.

Per our README, here is the differentiation between presentational and container components:

```
### Presentational Components

Presentational components do NOT fetch data to populate themselves.  They are the underlying UI building blocks of the views and connected components.  These are things such as buttons, cards, and a display for a single survey task.

### Container Components

Container components fetch data via the MyDataHelps.js SDK.  These components will only function if used in a view inside MyDataHelps or via manually passing an access token to the SDK.
```

Since the `Goal` component fetches data for itself using the MDH.js SDK (among other sources), I think it fits best with the container components.

> For example, this could not be dropped into vb without a wrapper. Unless I am missing something?

The `Goal` component can be dropped into the ViewBuilder anywhere and it should function appropriately.  A user may wish to place it within a `Card` or a `Section` component, but that wouldn't be necessary.

File: package.json
Comment (greinard) [reply package.json]: Will do.

File: src/helpers/single-value-providers.ts
Comment (greinard) [reply src/helpers/single-value-providers.ts]: Done.  Thanks for the suggestion.

**Comment by greinard** [comment]
I think I have addressed all of the feedback thus far.  Please let me know if I have not alleviated a concern or if you would like me to address anything else/further.

**Comment by greinard** [comment]
I have reverted the snapshot version numbers.  I would like to merge this today, if possible.

Are there any further objections or requests for delay?

Thanks!

**Review by reneefromhold — APPROVED** [review]

**Pull request merged**
